FinSearchUnified/Subscriber/Frontend.php

```diff
@@ -54,13 +53,17 @@ public function onFrontendPreDispatch(Enlight_Event_EventArgs $args)
         /** @var Request $request */
         $request = $subject->Request();
 
+        $controllerName = $request->getControllerName();
+        $moduleName = $request->getModuleName();
+
         if ($this->isSearchPage($request)) {
             Shopware()->Session()->offsetSet('isSearchPage', true);
             Shopware()->Session()->offsetSet('isCategoryPage', false);
         } elseif ($this->isCategoryPage($request)) {
             Shopware()->Session()->offsetSet('isCategoryPage', true);
             Shopware()->Session()->offsetSet('isSearchPage', false);
-        } elseif ($this->isManufacturerPage($request)) {
+        } elseif ($this->isManufacturerPage($request) || $moduleName == 'backend' || ($controllerName !== 'listing'
```

Comment:
* Move everything `$controllerName` related to the next line, just to make it visually clear that they belong together.
* Use `===`. Always.